torch_geometric/explain/config.py

```diff
@@ -75,6 +76,9 @@ class ExplainerConfig(CastMixin):
 
                 - :obj:`"both"`: Will mask both the node and its attributes..
 
+                - :obj:`"common_attributes"`: Will share the attribute mask
+                    across all nodes.
+
```

Comment:
What is the difference between common_attributes and either attributes or both? My understanding is common_attributes is (N, F) like both but is just attributes repeated N times. If that is the case, can it be extracted out of the individual explanation methods (treat independent of GNNExplainer logic)?

Reply:
The idea is to have users go through the `Explainer` class, and not the `ExplainerAlgorithm` class. This choice might also be useful for other explanation methods. 

I have reshaped the output to be a node_feat_mask so that any downstream task does not have to worry about how the feature mask of the nodes was computed, but just take it as a node_feat_mask. Maybe adding a new type of mask might make sens, but it would be additional to the node_feat_mask ?

Reply:
```python
if explainer_config.node_mask_type == MaskType.common_attributes:
            node_mask = self.node_mask.detach().sigmoid().squeeze(0)
            node_mask = self._reshape_common_attributes(node_mask, num_nodes)
```

In this snippet from GNNExplainer, the logic seems completely separable from the specific method. Could this maybe be moved outo f GNNExplainer and into ExplainerAlgorithm?

Reply:
The `_reshape_common_attributes` function is already in the base class, I'm not sure I see what you mean.